endpoint/endpoint_test.go

```diff
@@ -216,3 +216,191 @@ func TestIsOwnedBy(t *testing.T) {
 		})
 	}
 }
+
+func TestDuplicatedFilterEndpointsByOwnerID(t *testing.T) {
+	foo1 := &Endpoint{
+		DNSName:    "foo.com",
+		RecordType: RecordTypeA,
+		Labels: Labels{
+			OwnerLabelKey: "foo",
+		},
+	}
+	foo2 := &Endpoint{
+		DNSName:    "foo.com",
+		RecordType: RecordTypeCNAME,
+		Labels: Labels{
+			OwnerLabelKey: "foo",
+		},
+	}
+	foo3 := &Endpoint{
+		DNSName:    "foo.com",
+		RecordType: RecordTypeA,
+		Labels: Labels{
+			OwnerLabelKey: "foo",
+		},
+	}
+	foo4 := &Endpoint{
+		DNSName:    "foo.com",
+		RecordType: RecordTypeCNAME,
+		Labels: Labels{
+			OwnerLabelKey: "foo",
+		},
+	}
+	bar := &Endpoint{
+		DNSName:    "foo.com",
+		RecordType: RecordTypeA,
+		Labels: Labels{
+			OwnerLabelKey: "bar",
+		},
+	}
+
+	type args struct {
+		ownerID string
+		eps     []*Endpoint
+	}
+	tests := []struct {
+		name string
+		args args
+		want []*Endpoint
+	}{
+		{
+			name: "filter values",
+			args: args{
+				ownerID: "foo",
+				eps: []*Endpoint{
+					foo1,
+					foo2,
+					foo3,
+					foo4,
+					bar,
+				},
+			},
+			want: []*Endpoint{
+				foo1,
+				foo2,
```

Comment:
This result looks wrong. You can not have CNAME and A record with the same name.
Another problem is that if you have multiple IPs in your ingress status, you will want to set multiple A records for the same name.

Reply:
@szuecs Thank you for your review. I was taking as example the unit test on https://github.com/kubernetes-sigs/external-dns/blob/master/endpoint/endpoint_test.go#L120-L174. About the ingress status with multiple IPs issue, does that mean that we don't have unit tests for that use case and that's why I got a pass on the tests?